src/components/Map/Alerts/HazardLayer.tsx

```diff
@@ -0,0 +1,41 @@
+import { useMemo } from 'react';
+import MarkerClusterGroup from 'react-leaflet-cluster';
+
+import LegendContainer from '@/components/Legend/LegendContainer';
+import { MAP_MAX_ZOOM } from '@/domain/constant/Map';
+import { HazardType } from '@/domain/enums/HazardType';
+import { useHazardQuery } from '@/domain/hooks/alertHooks';
+import HazardOperations from '@/operations/alerts/HazardOperations';
+
+import { HazardMarker } from './HazardMarker';
+
+export function HazardLayer() {
+  const { data, isPending } = useHazardQuery();
+  const hazardsByType = useMemo(() => HazardOperations.sortHazardsByType(data), [data]);
+
+  return (
+    <>
+      <div className="absolute bottom-6 right-8 z-9999">
+        <LegendContainer loading={isPending || !data} items={HazardOperations.generateHazardLegend()} />
+      </div>
```

Comment:
this is overridden by other legend containers, so this won't be the final solution, but at least it's testable this way